charts/telemetry-collector/templates/deployment.yaml

```
@@ -0,0 +1,33 @@
+{{- include
+      "accelleran.common.deployment"
+      (fromYaml (include "accelleran.telemetry-collector.deployment.args" $))
+-}}
+
+{{- define "accelleran.telemetry-collector.deployment.args" -}}
+{{- $ := . -}}
+{{- $fullname := include "accelleran.common.fullname" (dict "top" $) | trunc 40 -}}
+
+top:
+  {{ $ | toYaml | nindent 2 }}
+
+volumes:
+- name: {{ $fullname }}-config-volume
+  emptyDir: {}
+- name: {{ $fullname }}-config
+  configMap:
+    name: {{ $fullname }}-config
+
+
+volumeMounts:
+- name: {{ $fullname }}-config
+  mountPath: /app/config.yaml
+  subPath: "config.yaml"
+
+env:
+  - name: INTERNAL_INFLUXDB_TOKEN
+    valueFrom:
+      secretKeyRef:
+        name: "{{ $.Release.Name }}-influxdb2-auth"
```

We should inject this in some way through the values file (same reason as the influxdb url: it's not deployed as part of this chart).

So for example it could become something like this (`tpl` to allow injecting the `.Release.Name` from the drax chart):

```suggestion
        name: "{{ tpl $.Values.influxdbAuthSecret $ }}"
```